.github/workflows/ci_macos.yml

```diff
@@ -1,132 +1,84 @@
-name: CI on macOS
+name: macOS
 
 on:
   push:
-    branches:
-      - 'main'
   pull_request:
   workflow_dispatch:
 
 concurrency:
   group: macos-${{ github.event.pull_request.number || github.ref }}
-  cancel-in-progress: true
+  cancel-in-progress: ${{ github.event_name != 'push' }}
 
 env:
-  CMAKE_VERSION: 3.16.9
-  SEQAN3_NO_VERSION_CHECK: 1
   SHARG_NO_VERSION_CHECK: 1
   TZ: Europe/Berlin
 
 defaults:
   run:
-    shell: bash -ex {0}
+    shell: bash -Eeuxo pipefail {0}
 
 jobs:
   build:
     name: ${{ matrix.name }}
     runs-on: macos-12
     timeout-minutes: 120
     strategy:
-      fail-fast: true
+      fail-fast: false
```

Comment:
I think in the app template we should fail-fast....

Reply:
But it's annoying when one fails, but you don't know whether the others also fail (compiler-specific)

Reply:
If the includes have proper guards, it should still compile. I'll move it up and check the other files. Looks like it was even wrong before this pr :D